frontend/src/lib-common/utils/usePendingUserInput.ts

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { useEffect, useState } from 'react'
+
+export function usePendingUserInput<T>(
+  transform: (input: string) => T | null,
+  initialValue?: string
+) {
+  const [input, setInput] = useState(initialValue ?? '')
+  const [validated, setValidated] = useState<T>()
+
+  useEffect(() => {
+    const transformed = transform(input)
+
+    if (transformed) {
+      setValidated(transformed)
+    } else {
+      setValidated(undefined)
+    }
+  }, [input, transform])
+
+  return [input, setInput, validated] as [
```

Comment:
Can this type assertion be removed if you add just add a return type to the function?

Reply:
Yes – changed. 😄